backend/routes/download.js

```diff
@@ -10,20 +10,30 @@ import Group from '../models/group'
 import Image from '../models/image'
 import Show from '../models/show'
 import User from '../models/user'
-import { IMAGE_ENTRY } from '../constants'
+import { IMAGE_ENTRY, ADMIN } from '../constants'
 import config from '../config'
-import sequelize from '../config/sequelize';
+import sequelize from '../config/sequelize'
+import { parseToken } from '../helpers/jwt'
 
 const readFileAsync = promisify(fs.readFile)
 
 const IMAGE_DIR = config.get('upload:imageDir')
 
 const router = Router()
 
-router.route('/zips/:showId')
-  .get((req, res, next) => {
-    // TODO figure out how to do auth
+const ensureAdminDownloadToken = (req, res, next) => {
+  const token = req.query.token
+  parseToken(token, (err, decoded) => {
+    if (err || decoded.type !== ADMIN) {
+      res.status(401).send('{"message": "permission denied"}')
```

Comment:
I think the identifier here should be consistent with /routes/auth. In auth you use `error: message`. Your non-success status messages are created different ways in this file.

Reply:
good point, will do

Reply:
I realized that it makes no sense to render json here anyway, since this text will be displayed on a user's browser if they are not permitted to access this... I changed it to HTML in 88b6abf